regression-tests/tests/hello-new-string.ssl

```diff
@@ -0,0 +1,4 @@
+main cin cout =
```

Comment:
Above main, add an ADT definition for a list that uses Cons and Nil as data constructors.